src/models/chen2021Fitness/Chen2021FitnessWidget.tsx

```diff
@@ -0,0 +1,381 @@
+import { Widget } from '../../widgets/Widget';
+import { ZodQueryEncoder } from '../../helpers/query-encoder';
+import { SampleSelectorSchema } from '../../helpers/sample-selector';
+import * as zod from 'zod';
+import React, { useEffect, useState } from 'react';
+import { get } from '../../services/api';
+import {
+  Chen2021FitnessRequestSchema,
+  Chen2021FitnessResponseSchema,
+  ValueWithCISchema,
+} from './chen2021Fitness-types';
+import styled from 'styled-components';
+import { Col, Form, Button } from 'react-bootstrap';
+import { Plot } from '../../components/Plot';
+
+const ContainerPropsSchema = SampleSelectorSchema;
+type ContainerProps = zod.infer<typeof SampleSelectorSchema>;
+
+const ResultsPropsSchema = zod.object({
+  request: Chen2021FitnessRequestSchema,
+});
+type ResultsProps = zod.infer<typeof ResultsPropsSchema>;
+
+type Chen2021FitnessRequest = zod.infer<typeof Chen2021FitnessRequestSchema>;
+type Chen2021FitnessResponse = zod.infer<typeof Chen2021FitnessResponseSchema>;
+
+const getData = async (
+  params: Chen2021FitnessRequest,
+  signal: AbortSignal
+): Promise<Chen2021FitnessResponse | null> => {
+  const mutationsString = params.mutations.join(',');
+  const url =
```

Comment:
[`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) is designed exactly for building URLs like this. You could use `new URLSearchParams(params)` to set most values and then `.set(...)` and `.delete(...)` to handle the few special fields.